lib/model.py

```diff
@@ -23,44 +40,64 @@ def is_installed(executable_cmd):
     return False
 
 
+def djoin(*tup):
+    """Convenience method to join strings with dots"""
+    return '.'.join(tup)
+
+
+def flatten(list_of_lists):
+    """
+    Return a list from a list of lists
+    :param list[list[Value]] list_of_lists:
+    :rtype: list[Value]
+    """
+    return [el for sublist in list_of_lists for el in sublist]
+
+
 def _resolve_str_variable(variable, file_groups):
     """
     String variables are when variable.points_to is a string
-    This indicates imports that we delayed processing for
+    This happens ONLY when we have imports that we delayed processing for
     This function looks through all processed files to see if we can find a match
 
     :param Variable variable:
     :param list[Group] file_groups:
     :rtype: Node|str
     """
+    has_known = False
 
     for file_group in file_groups:
+        # Check if any top level node in the other file matches the import
         for node in file_group.nodes:
-            if file_group.token + '.' + node.token == variable.points_to:
+            if djoin(file_group.token, node.token) == variable.points_to:
                 return node
+        # Check if any top level class in the other file matches the import
         for group in file_group.subgroups:
-            if file_group.token + '.' + group.token == variable.points_to \
+            if djoin(file_group.token, group.token) == variable.points_to \
                and group.get_constructor():
                 return group.get_constructor()
-    return "UNKNOWN_MODULE"  # Default indicates we must skip
+        if file_group.token == variable.points_to.split('.', 1)[0]:
+            has_known = True
+    if has_known:
+        return OWNER_CONST.KNOWN_MODULE
+
+    return OWNER_CONST.UNKNOWN_MODULE  # Default indicates we must skip
 
 
 class BaseLanguage(abc.ABC):
     """
     Languages are individual implementations for different dynamic languages.
-    This will eventually be the superclass of Python, Javascript, PHP, and Ruby.
+    This is the superclass of Python, Javascript, PHP, and Ruby.
     Every implementation must implement all of these methods.
     For more detail, see the individual implementations.
-    (Note that the 'Tree' parameter / rtype is generic and will be a different
-     type for different languages. In Python, it is an ast)
+    Note that the 'Tree' type is generic and will be a different
+    type for different languages. In Python, it is an ast.AST.
     """
 
-    @property
+    @staticmethod
     @abc.abstractmethod
-    def RESERVED_KEYWORDS(self):
-        """
-        :rtype: list[str]
-        """
+    def assert_dependencies():
+        """"""
 
     @staticmethod
     @abc.abstractmethod
```

Comment:
The abstract `def get_tree(filename)` method should take two arguments `filename` and `source_type`. The `source_type` value is used by the Javascript class.